.github/workflows/release-homebrew.yaml

```diff
@@ -0,0 +1,164 @@
+      - name: Clean up previous files
+        run: |
+          sudo rm -rf /opt/finch
+          sudo rm -rf ~/.finch
+          if pgrep '^qemu-system'; then
+            sudo pkill '^qemu-system'
+          fi
+          if pgrep '^socket_vmnet'; then
+            sudo pkill '^socket_vmnet'
+          fi
+          cd $(brew --repo homebrew/cask) 
+          brew uninstall --cask ./Casks/finch.rb || true
+      - name: Set up Homebrew
+        id: set-up-homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+      - name: Bump local cask version
+        run: |
+          brew update-reset 
+          cd $(brew --repo homebrew/cask)
+          echo "Update the local Finch cask with version ${FINCH_VERSION}"
+          brew bump-cask-pr --write-only -f --version=${FINCH_VERSION} finch
+      - name: Silently install
+        run: |
+          cd $(brew --repo homebrew/cask)
+          brew install --cask ./Casks/finch.rb
+      - name: Run e2e tests
+        run: INSTALLED=true make test-e2e
+      - name: Silently uninstall
+        if: ${{ always() }}
+        run: |
+          cd $(brew --repo homebrew/cask)
+          brew uninstall --cask ./Casks/finch.rb
+
+  macos-amd64-test-installer:
+    needs: get-latest-tag
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          [
+            [self-hosted, macos, amd64, 10.15, release],
+            [self-hosted, macos, amd64, 11.7, release],
+            [self-hosted, macos, amd64, 12.6, release],
+            [self-hosted, macos, amd64, 13.0, release],
+          ]
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 60
+    env:
+      ACCESS_TOKEN: ${{ secrets.FINCH_BOT_TOKEN }}
+      FINCH_TAG: ${{ needs.get-latest-tag.outputs.tag }}
+      FINCH_VESRION: ${{ needs.get-latest-tag.outputs.version }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ env.FINCH_TAG }}
+      - name: Clean up previous files
+        run: |
+          sudo rm -rf /opt/finch
+          sudo rm -rf ~/.finch
+          if pgrep '^qemu-system'; then
+            sudo pkill '^qemu-system'
+          fi
+          if pgrep '^socket_vmnet'; then
+            sudo pkill '^socket_vmnet'
+          fi 
+          cd $(brew --repo homebrew/cask) 
+          brew uninstall --cask ./Casks/finch.rb || true
+      - name: Set up Homebrew
+        id: set-up-homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+      - name: Bump local cask version
+        run: |
+          brew update-reset 
+          cd $(brew --repo homebrew/cask)
+          echo "Update the local Finch cask with version ${FINCH_VERSION}"
+          brew bump-cask-pr --write-only -f --version=${FINCH_VERSION} finch
+      - name: Silently install
+        run: |
+          cd $(brew --repo homebrew/cask)
+          brew install --cask ./Casks/finch.rb
+      - name: Run e2e tests
+        run: INSTALLED=true make test-e2e
+      - name: Silently uninstall
```

Comment:
Not related with PR, but seems we didn't have installer test such as after uninstall, e.g., need to assert directory is removed after uninstall

Reply:
I can investigate whether we can add it to the workflow.

Reply:
Added check for uninstallation.